src/algebra/monoid_algebra/basic.lean

```diff
@@ -1503,6 +1533,18 @@ finset.induction_on s rfl $ λ a s has ih, by rw [prod_insert has, ih,
 
 end
 
+/--  An additive homomorphism `f : G →+ H` induces a `k`-algebra homomorphism
+`add_monoid_alg_hom_map k A f : add_monoid_algebra A G →ₐ[k] add_monoid_algebra A H`. -/
+def add_monoid_alg_hom_map (k A : Type*) [comm_semiring k] [semiring A] [algebra k A]
+  [add_monoid G] {H : Type*} [add_monoid H]
+  (f : G →+ H) : add_monoid_algebra A G →ₐ[k] add_monoid_algebra A H :=
+lift_nc_alg_hom single_zero_alg_hom ((of A H).comp f.to_multiplicative) $
+λ a g, add_monoid_algebra.single_mul_single.trans (by
+{ simp only [zero_add, mul_one, single_one_right_add_hom_apply, single_zero_alg_hom_apply,
+    ring_hom.to_fun_eq_coe, single_zero_ring_hom_apply, add_monoid_hom.to_fun_eq_coe,
+    single_add_hom_apply, single_mul_single, add_zero, one_mul],
+  refl })
```

Comment:
Much shorter as:
```suggestion
{ commutes' := λ r, map_domain_single.trans $ by simp,
  ..map_domain_ring_hom A f}
```

Note that this `commutes'` field should probably be a standalone `simp` lemma called `map_domain_algebra_map`

Reply:
Ok, I extracted the lemma, made it simp and replaced the proof of `commutes'` as an application of the lemma.